docs/book/src/reference/webhook-for-core-types.md

```diff
@@ -58,13 +88,175 @@ If you need a client and/or decoder, just pass them in at struct construction ti
 
 ```go
 mgr.GetWebhookServer().Register("/mutate-v1-pod", &webhook.Admission{
-	Handler: &podAnnotator{
-		Client:   mgr.GetClient(),
-		decoder:  admission.NewDecoder(mgr.GetScheme()),
-	},
+    Handler: &podAnnotator{
+        Client:   mgr.GetClient(),
+        decoder:  admission.NewDecoder(mgr.GetScheme()),
+    },
 })
 ```
 
+
+## By using Custom interfaces instead of Handle
+
+### Update your webhook:
+
+**Example**
+
+Following an example of its implementation.
+
+```go
+package v1
+
+import (
+  "context"
+  "fmt"
+
+  corev1 "k8s.io/api/core/v1"
+  "k8s.io/apimachinery/pkg/runtime"
+  ctrl "sigs.k8s.io/controller-runtime"
+  logf "sigs.k8s.io/controller-runtime/pkg/log"
+  "sigs.k8s.io/controller-runtime/pkg/webhook"
+  "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// log is for logging in this package.
+var podlog = logf.Log.WithName("pod-webhook")
+
+// SetupWebhookWithManager sets up the webhook with the manager for both the defaulter and validator
+func (r *corev1.Pod) SetupWebhookWithManager(mgr ctrl.Manager) error {
+  return ctrl.NewWebhookManagedBy(mgr).
+    For(r).
+    WithValidator(&PodCustomValidator{}).
+    WithDefaulter(&PodCustomDefaulter{}).
+    Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io,admissionReviewVersions=v1
```

Comment:
IIRC I had to add something like https://github.com/mythi/kubebuilder/blob/2db6ee6f869b69ba46501b734e50c41a438bc5d6/docs/book/src/reference/webhook-for-core-types.md?plain=1#L47-L50 to get it working.

Reply:
The same marker is in the line 47 
What exactly is missing could you please let me know?

Reply:
I had this clarification added: "(for webhooks the `path` is of format `/mutate-<group>-<version>-<kind>`. Since this documentation uses `Pod` from the core API group, the group needs to be an empty string)"

the marker is: `/mutate--v1-pod`

Reply:
done good catcher